containerd: set `LimitNOFILE` on the service

NicolasT opened the pull request:
The default 'open files' limit, 1024, is way too low for some of our
applications. Since there's no way to bump this limit from within a
container, and the artificial limit of 1024 file descriptors stems from
an era where FDs were costly, bump the limit to a more reasonable (in
2019) default.

Manual test:

```
[root@bootstrap ~]# kubectl taint node bootstrap node-role.kubernetes.io/bootstrap-
node/bootstrap untainted
[root@bootstrap ~]# kubectl taint node bootstrap node-role.kubernetes.io/infra-
node/bootstrap untainted
[root@bootstrap ~]# kubectl run --restart=Never --image=busybox shell -- sleep 9999
pod/shell created
[root@bootstrap ~]# kubectl exec -ti shell -- sh -c 'ulimit -n'
1048576
```

Fixes: #1785
See: https://github.com/scality/metalk8s/issues/1785
See: https://github.com/containerd/containerd/issues/3201
See: https://github.com/containerd/containerd/pull/3202

NicolasT committed:
containerd: set `LimitNOFILE` on the service

The default 'open files' limit, 1024, is way too low for some of our
applications. Since there's no way to bump this limit from within a
container, and the artificial limit of 1024 file descriptors stems from
an era where FDs were costly, bump the limit to a more reasonable (in
2019) default.

Manual test:

```
[root@bootstrap ~]# kubectl taint node bootstrap node-role.kubernetes.io/bootstrap-
node/bootstrap untainted
[root@bootstrap ~]# kubectl taint node bootstrap node-role.kubernetes.io/infra-
node/bootstrap untainted
[root@bootstrap ~]# kubectl run --restart=Never --image=busybox shell -- sleep 9999
pod/shell created
[root@bootstrap ~]# kubectl exec -ti shell -- sh -c 'ulimit -n'
1048576
```

Fixes: #1785
See: https://github.com/scality/metalk8s/issues/1785
See: https://github.com/containerd/containerd/issues/3201
See: https://github.com/containerd/containerd/pull/3202

bert-e commented:
# Hello nicolast,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @sayf-eddine-scality

* @thomasdanan

* @carlito-scality

* @teddyandrieux

* @nootal

* @lucieleonard

* @gdemonet

* @jbertran

* @nicolast

* @monpote

* @tcarmet

* @slaperche-scality

NicolasT commented:
~Need to test this first.~

NicolasT commented:
I'll approve once @fferrandis signed off in the issue :)

NicolasT committed:
kubernetes, salt: raise `salt-run` readiness probe timeout

Oddly enough, the `ulimit` change of the next commit (cfr. #1785) causes
the readiness probe of the `salt-master` container (which runs `salt-run
event.send test` in the container) to timeout (i.e., not return
successfully within the currently configured timeout of 2s).

Not entirely sure why ¯\_(ツ)_/¯

However, on a somewhat loaded Salt system, such 2s timeout is way too
low anyway, at which point the system (e.g., our `ext_pillar` modules
that retrieve *Endpoints* of various *Services*, including the
`salt-master` one...) could start to behave oddly (which is how this was
detected in the first place).

As such, bumping the timeout to a (hopefully) more reasonable 10
seconds.

See: #1785
See: https://github.com/scality/metalk8s/issues/1785

fferrandis commented on salt/metalk8s/container-engine/containerd/files/50-metalk8s.conf:
```diff
@@ -0,0 +1,4 @@
+[Service]
+# See https://github.com/containerd/containerd/issues/3201
+# See https://github.com/containerd/containerd/pull/3202
+LimitNOFILE=1048576
```
<3

fferrandis commented:
Thank you team !

NicolasT commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye nicolast.
